pymeasure/instruments/anritsu/anritsuMS464xB.py

```diff
@@ -0,0 +1,912 @@
+# THE SOFTWARE.
+#
+import logging
+
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import (
+    strict_discrete_set,
+    strict_range
+)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class AnritsuMS464xB(Instrument):
+    """ A class representing the Anritsu MS464xB Vector Network Analyzer (VNA) series.
+
+    This familly consists of the MS4642B, MS4644B, MS4645B, and MS4647B, which are represented in
+    their respective classes (:class:`~.AnritsuMS4642B`, :class:`~.AnritsuMS4644B`,
+    :class:`~.AnritsuMS4645B`, :class:`~.AnritsuMS4647B`), that only differ in the available
+    frequency range.
+
+    Can contain up to 16 instances of :class:`~.MeasurementChannel` (depending on the configuration
+    of the instrument), that are accessible via the `channels` dict or directly via `ch_` + the
+    channel number.
+
+    :param active_channels: defines the number of active channels (default=16); if active_channels
+        is "auto", the instrument will be queried for the number of active channels.
+    :type active_channels: int (1-16) or str ("auto")
+    :param installed_ports: defines the number of installed ports (default=4); if "auto" is
+        provided, the instrument will be queried for the number of ports
+    :type installed_ports: int (1-4) or str ("auto")
+    :param traces_per_channel: defines the number of traces that is assumed for each channel
+        (between 1 and 16); if not provided, the maximum number is assumed; "auto" is provided,
+        the instrument will be queried for the number of traces of each channel.
+    :type traces_per_channel: int (1-16) or str ("auto") or None
+    """
+    CHANNELS_MAX = 16
+    TRACES_MAX = 16
+    PORTS = 4
+    TRIGGER_TYPES = ["POIN", "SWE", "CHAN", "ALL"]
+
+    FREQUENCY_RANGE = [1E7, 7E10]
+
+    SPARAM_LIST = ["S11", "S12", "S21", "S22",
+                   "S13", "S23", "S33", "S31",
+                   "S32", "S14", "S24", "S34",
+                   "S41", "S42", "S43", "S44", ]
+    DISPLAY_LAYOUTS = ["R1C1", "R1C2", "R2C1", "R1C3", "R3C1",
+                       "R2C2C1", "R2C1C2", "C2R2R1", "C2R1R2",
+                       "R1C4", "R4C1", "R2C2", "R2C3", "R3C2",
+                       "R2C4", "R4C2", "R3C3", "R5C2", "R2C5",
+                       "R4C3", "R3C4", "R4C4"]
+
+    def __init__(self, adapter,
+                 name="Anritsu MS464xB Vector Network Analyzer",
+                 active_channels=16,
+                 installed_ports=4,
+                 traces_per_channel=None,
+                 **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            timeout=10000,
+            **kwargs,
+        )
+
+        self.PORTS = installed_ports if installed_ports != "auto" else self.number_of_ports
+
+        number_of_channels = active_channels if active_channels != "auto" else None
```

Comment:
For me it is easier to read, if you have an equality comparison:
~~~suggestion
        self.PORTS = self.number_of_ports if installed_ports == "auto" else installed_ports

        number_of_channels = None if active_channels == "auto" else active_channels
~~~

Reply:
A, yeah, that is indeed clearer